client/js/app/components/explorer/visualization/keen_viz.js

```diff
@@ -35,6 +35,9 @@ var KeenViz = React.createClass({
   // ***********************
 
   shouldComponentUpdate: function(nextProps, nextState) {
+    if (this.props.model.metadata.visualization.chart_type === "table") {
```

Comment:
`KeenViz` won't update if `extractionFields` are changed (it only updates when the viz type changes), which means when a new extraction field is selected, the table won't update. The code above will update the viz if an extraction field is selected.